presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2636,7 +2638,7 @@ private boolean isPushdownFilterEnabled(ConnectorSession session, ConnectorTable
         boolean pushdownFilterEnabled = HiveSessionProperties.isPushdownFilterEnabled(session);
         if (pushdownFilterEnabled) {
             HiveStorageFormat hiveStorageFormat = getHiveStorageFormat(getTableMetadata(session, tableHandle).getProperties());
-            if (hiveStorageFormat == ORC || hiveStorageFormat == DWRF) {
+            if (hiveStorageFormat == ORC || hiveStorageFormat == DWRF || hiveStorageFormat == PARQUET && isParquetPushdownFilterEnabled(session)) {
```

Comment:
are you missing parentheses?

Reply:
`&&` is higher priority than `||` therefore this line is equal to 
```
if (hiveStorageFormat == ORC || hiveStorageFormat == DWRF || (hiveStorageFormat == PARQUET && isParquetPushdownFilterEnabled(session)))
```
I think this is correct, since PARQUET selective readers are not implemented yet, and if hive.pushdown_filter_enabled is set true in order to run ORC queries fast, Parquet queries would fail. Therefore we need a second session property on Parquet and enables the pushdown only when both conditions are met.

Reply:
I see. This is very subtle and hard to read. I think adding parenthesis will be clearer.